lib/Model/Announcement.php

```diff
@@ -59,13 +67,28 @@ class Announcement extends Entity {
 	/** @var int */
 	protected $allowComments;
 
+	/** @var int */
+	protected $scheduleTime;
+
+	/** @var int */
+	protected $deleteTime;
+
+	/** @var string */
+	protected $groups;
+
+	/** @var int */
+	protected $notTypes;
+
 	public function __construct() {
 		$this->addType('time', 'int');
 		$this->addType('user', 'string');
 		$this->addType('subject', 'string');
 		$this->addType('message', 'string');
 		$this->addType('plainMessage', 'string');
 		$this->addType('allowComments', 'int');
+		$this->addType('scheduleTime', 'int');
+		$this->addType('deleteTime', 'int');
+		$this->addType('notTypes', 'int');
```

Comment:
groups is not registered here, it's string on the database?

Reply:
Yes, it has to be, because when the announcement will be, well, announced in the future, it needs to notifiy the groups, so they need to be stored somewhere

Reply:
yeah I more meant to register all fields, even if they have the "default" type